### DISCUSSION TITLE
Seeking help with l5-swagger on a bref serverless api

### DISCUSSION THREAD

**Discussion by LoreSco** [DarkaOnLine/L5-Swagger#599] [Q&A]
Hi everyone,
I'm trying to add the l5swagger to my serverless api built using bref and laravel 10.
Locally (of course XD)  it works fine, when deploying it to AWS instead I'm facing an issue with the assets:

The view is looking to retrieve the assets from

`https:://mydomain.execute-api.us-east-1.amazonaws.com/docs/assets/swagger-ui-bundle.js`

while the real address has also the stage. I'm struggling to find which config I need to change to add the app stage as prefix to make it match :)

`https:://mydomain.execute-api.us-east-1.amazonaws.com/dev/docs/assets/swagger-ui-bundle.js`

Thanks

**Comment by umer-nxb** [comment]
@LoreSco found any solutions?

**Reply by LoreSco** [reply]
I solved with a workaround creating an index.html that uses the `<script src="https://unpkg.com/swagger-ui-dist@5.11.0/swagger-ui-bundle.js" crossorigin></script>` and generating the api-docs.json file in that same folder

**Reply by umer-nxb** [reply]
Interesting find.
I have just found another solutions based on your confirmation that actual issue with obtaining the path to api-docs.json. I could modify the 

```
// config/l5-swagger.php

defaults.paths.docs  => "/urlprefix/" . storage_path("api-docs");

```
 but that would change the docs generatation path as well. so I kept it as provided by published config i.e `defaults.paths.docs  =>  storage_path("api-docs");` but in the generated blade view in `resources/views/vendor/l5-swagger/index.blade.php" I have added url prefix like this:
 
 ```
 // Build a system
        const ui = SwaggerUIBundle({
            dom_id: '#swagger-ui',
            url: "/my-url-prefix{!! $urlToDocs !!}",
           // ... remaining config
        })
        
 ```
 
 Now, the command `php artisan l5-swagger:generate` is generating the docs in $urlToDocs but I am able to serve it on /my-url-prefix/$urlToDocs
 
 
 Also, for other assets like css and js bundle, I had to prefix the href and src with "/my-url-prefix"
 
 Hope that someone would save great time.
 
 But, @DarkaOnLine  I request that configuration for generating the docs should be separate from the configuration to serve them.  Or something better to handle  discussion#599.
